sdk/loadtestservice/Azure.Developer.LoadTesting/src/FileUploadOperation.cs

```diff
@@ -0,0 +1,133 @@
+        private Response _response;
+        private BinaryData _value;
+        private readonly string _testId;
+        private readonly string _fileName;
+        private readonly LoadTestAdministrationClient _client;
+        private readonly List<string> _terminalStatus = new()
+            {
+                "VALIDATION_SUCCESS",
+                "VALIDATION_FAILURE",
+                "VALIDATION_NOT_REQUIRED"
+            };
+
+        /// <summary>
+        /// Value.
+        /// </summary>
+        public override BinaryData Value => _value;
+
+        /// <summary>
+        /// HasValue
+        /// </summary>
+        public override bool HasValue => _completed;
+
+        /// <summary>
+        /// Id.
+        /// </summary>
+        public override string Id { get; }
+
+        /// <summary>
+        /// HasCompleted
+        /// </summary>
+        public override bool HasCompleted => _completed;
+
+        /// <summary>
+        /// FileUploadOperation.
+        /// </summary>
+        protected FileUploadOperation() { }
+
+        /// <summary>
+        /// FileUploadOperation.
+        /// </summary>
+        public FileUploadOperation(string testId, string fileName, LoadTestAdministrationClient client, Response initialResponse)
+        {
+            _testId = testId;
+            _fileName = fileName;
+            _client = client;
+            Id = $"{_testId}/{_fileName}";
+            _completed = false;
+            if (initialResponse != null)
+            {
+                _response = initialResponse;
+                _value = _response.Content;
+                GetCompletionResponse();
+            }
+        }
+
+        /// <summary>
+        /// GetRawResponse.
+        /// </summary>
+        public override Response GetRawResponse()
+        {
+            return _response;
+        }
+
+        /// <summary>
+        /// UpdateStatus.
+        /// </summary>
+        public override Response UpdateStatus(CancellationToken cancellationToken = default)
+        {
+            if (_completed)
+            {
+                return GetRawResponse();
+            }
+
+            _response = _client.GetTestFile(_testId, _fileName);
+            _value = _response.Content;
+
+            return GetCompletionResponse();
+        }
+
+        /// <summary>
+        /// UpdateStatusAsync.
+        /// </summary>
+        public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default)
+        {
+            if (_completed)
+            {
+                return GetRawResponse();
+            }
+
+            _response = await _client.GetTestFileAsync(_testId, _fileName).ConfigureAwait(false);
+            _value = _response.Content;
+
+            return GetCompletionResponse();
+        }
+
+        private Response GetCompletionResponse()
+        {
+            string fileValidationStatus;
+
+            JsonDocument jsonDocument = JsonDocument.Parse(_value.ToString());
```

Comment:
I think ToMemory is better from a performance perspective.

```suggestion
            JsonDocument jsonDocument = JsonDocument.Parse(_value.ToMemory());
```